pkg/lease/service.go

```diff
@@ -91,6 +91,7 @@ func (a *Service) Save(data *Lease) error {
 	} else {
 		lastModifiedOn = data.LastModifiedOn
 		data.LastModifiedOn = &now
+		data.StatusModifiedOn = &now
```

Comment:
@AmudaPalani I don't think this is quite what we want.

The `LeaseStatusModifiedOn` field should only be updated if the `LeaseStatus` field changes.
Unfortunately, we have no way within the `Save()` method to know if the `LeaseStatus` field has been updated (we only have the new lease object, not the old one).

I'm not sure how best to approach this --we may need to set this timestamp  in some of the higher-level service methods (eg. `Create()` / `Delete()`). What do you think?

Reply:
This line 92-94 will be executed in two scenarios only:
a) Update a lease status from active to inactive (Decommission).
b) Update a lease status from inactive to active (Same account getting leased).
In both these scenarios, we want to update LeaseStatusModifiedOn.